### ISSUE TITLE
S3 Bucket HTTP access false positive

### ISSUE THREAD

**Issue by rcalcam** [Checkmarx/kics#5763]
Found a bug? You're welcome to [![GitHub Discussions](https://img.shields.io/badge/chat-discussions-blue.svg?style=flat-square)](https://github.com/Checkmarx/kics/discussions)

- **Please make sure to:**
  - Describe in details what the problem is
 Having an explicit "DENY" policy within a statement without 
``` 
"Condition": {
        "Bool": {
          "aws:SecureTransport": "false"
        }
      }
``` 
will result in a false positive stating that the bucket allows HTTP traffic, even if there is another explicit DENY statement which includes this condition
  - Attach a log file with relevant data preferably in DEBUG level (`--log-level=DEBUG`)
  N/A
  - Attach the scanned sample files, anonymize the data if the original file cannot be provided
  Example policy:
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "AllowSSLRequestsOnly",
      "Action": "s3:*",
      "Effect": "Deny",
      "Resource": [
        "${bucket_arn}",
        "${bucket_arn}/*"
      ],
      "Condition": {
        "Bool": {
          "aws:SecureTransport": "false"
        }
      },
      "Principal": "*"
    },
    {
      "Effect": "Deny",
      "Sid": "DenyActionsFromOutsideVpc",
      "Principal": "*",
      "Action": [
        "s3:*
      ],
      "Resource": [
        "arn:aws:s3:::${bucket_name}",
        "arn:aws:s3:::${bucket_name}/*"
      ],
      "Condition": {
        "StringNotLike": {
          "aws:SourceVpc": "${vpc_id}"
        }
      }
    }
  ]
}
```

### Expected Behavior

(Which results are expected from KICS?)
Policy should check if any one of the deny policies within the statement contain the condition to reject non-HTTPS connection

### Actual Behavior
The policy appears to check that each one of the deny statements contains the condition to reject non-HTTPS connection

### Steps to Reproduce the Problem

(Command line arguments and flags used)

1.  Create an s3 bucket
2.  Append policy with multiple deny statements - at least one of which denies non-https traffic (includes condition block) and at least one of which does not

### Specifications
N/A

**Comment by lrstanley** [comment]
To add additional context, this is related to the [terraform/aws/s3_bucket_policy_accepts_http_requests](https://github.com/Checkmarx/kics/blob/master/assets/queries/terraform/aws/s3_bucket_policy_accepts_http_requests/query.rego) query. All of the current negative tests for that query have only a single `Deny` statement, so it'd be good to get one that has multiple.

Though, testing this locally with a full example, I wasn't able to replicate (tested 1.5.10, 1.5.13 and 1.5.15 that I had locally installed):

```tf
resource "aws_s3_bucket" "b" {
  bucket = "my-tf-test-bucket"
}

resource "aws_s3_bucket_policy" "b" {
  bucket = aws_s3_bucket.b.id

  policy = <<EOF
{
    "Version": "2012-10-17",
    "Id": "MYBUCKETPOLICY",
    "Statement": [
      {
        "Sid": "DisableHTTP",
        "Effect": "Deny",
        "Principal": "*",
        "Action": "s3:*",
        "Resource": [
          "${aws_s3_bucket.b.arn}"
        ],
        "Condition": {
          "Bool": {
            "aws:SecureTransport": "false"
          }
        }
      },
      {
        "Sid": "DenyActionsFromOutsideVpc",
        "Effect": "Deny",
        "Principal": "*",
        "Action": [
          "s3:*"
        ],
        "Resource": [
          "${aws_s3_bucket.b.arn}",
          "${aws_s3_bucket.b.arn}/*"
        ],
        "Condition": {
          "StringNotLike": {
            "aws:SourceVpc": "${vpc_id}"
          }
        }
      }
    ]
}
EOF
}
```

**Comment by cxMiguelSilva** [comment]
Hi @rcalcam and @lrstanley, hope you are doing great!
I found an issue both in the Security Query Related and the sample @rcalcam provided.
@rcalcam your sample is missing an `"` in the action of the second statement that caused the Security Query to throw a result due to not being possible to unmarshal the JSON content of the policy.
Thank you for your input. 
The issue is being solved in #5832.

**Comment by rcalcam** [comment]
> Hi @rcalcam and @lrstanley, hope you are doing great!
> I found an issue both in the Security Query Related and the sample @rcalcam provided.
> @rcalcam your sample is missing an `"` in the action of the second statement that caused the Security Query to throw a result due to not being possible to unmarshal the JSON content of the policy.
> Thank you for your input.
> The issue is being solved in #5832.

HI @cxMiguelSilva , my mistake; it looks like I had a typo in the sample policy I provided. After doing some more testing, I believe the issue might be due to using jsonencode to pass a range of ip addresses. Please, check again with the following policy:
```
resource "aws_s3_bucket_policy" "access_policy" {
  bucket = aws_s3_bucket.b.id

  policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "AWS": "${aws_iam_role.r.arn}"
      },
      "Action": [
        "s3:PutObject",
        "s3:GetObject",
        "s3:DeleteObject",
        "s3:DeleteObjectVersion"
      ],
      "Resource": "arn:aws:s3:::${aws_s3_bucket.b.bucket}/*"
    },
    {
      "Effect": "Allow",
      "Principal": {
        "AWS": "${aws_iam_role.r.arn}"
      },
      "Action": [
        "s3:ListBucketVersions"
      ],
      "Resource": "arn:aws:s3:::${aws_s3_bucket.b.bucket}"
    },
    {
      "Sid": "AllowSSLRequestsOnly",
      "Action": "s3:*",
      "Effect": "Deny",
      "Resource": [
        "${aws_s3_bucket.b.arn}",
        "${aws_s3_bucket.b.arn}/*"
      ],
      "Condition": {
        "Bool": {
          "aws:SecureTransport": "false"
        }
      },
      "Principal": "*"
    },
    {
      "Effect": "Deny",
      "Sid": "DenyActionsFromOutsideVpcAndNetwork",
      "Principal": "*",
      "Action": [
        "s3:PutObject",
        "s3:GetObject",
        "s3:DeleteObject",
        "s3:ListBucketVersions",
        "s3:DeleteObjectVersion"
      ],
      "Resource": [
        "arn:aws:s3:::${aws_s3_bucket.b.bucket}",
        "arn:aws:s3:::${aws_s3_bucket.b.bucket}/*"
      ],
      "Condition": {
        "StringNotLike": {
          "aws:SourceVpc": "${data.aws_vpc.v.id}"
        },
        "NotIpAddress": {
          "aws:SourceIp": ${jsonencode(local.internal_ips)}
        }
      }
    }
  ]
}
POLICY
}
```
